### PR TITLE
PAAAReductor bug fix for non-parametric data

### PR DESCRIPTION
**Pull request by lbalicki**
In certain cases there are issues with properly dealing with complex sample data. This PR fixes the issues, adds a non-parametric data-driven reductor demo and improves the documentation. The code should now also be more in line with what the documentation of the `PAAAReductor` actually says.

### CODE REVIEW

File: src/pymordemos/dd_heat.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    err = fom - rom
+
+    n_w = 50
+    w = np.geomspace(ss[0]/2, ss[-1]*2, n_w)
+
+    fig, ax = plt.subplots(constrained_layout=True)
+    fom.transfer_function.mag_plot(w, ax=ax, label='FOM')
+    rom.mag_plot(w, ax=ax, label='ROM', linestyle='dashed')
+    err.mag_plot(w, ax=ax, label='Error', linestyle='dotted')
+    fig.suptitle(fr'Magnitude plot for {reductor_short_name}')
```
Comment (pmli) [inline src/pymordemos/dd_heat.py:46]:
```suggestion
    ax.set_title(fr'Magnitude plot for {reductor_short_name}')
```

**Review by pmli — COMMENTED** [review]
There is one suggestion below. Also, it would be good to document that `sampling_values` should be a list of 1D NumPy arrays (and maybe also assert that in the `__init__` method).

**Comment by lbalicki** [comment]
I made the changes and also allowed for more flexible inputs. In the non-parametric case we can now just call `PAAAReductor(s, fom)` instead of calling `PAAAReductor([s], fom)` if `s` is a numpy array with frequency samples.

File: src/pymor/reductors/aaa.py [hunk 2]
```diff
@@ -34,13 +34,16 @@ class PAAAReductor(BasicObject):
     sampling_values
         Values where sample data has been evaluated or the full-order model should be evaluated.
         Sampling values are represented as a nested list such that `sampling_values[i]` corresponds
```
Comment (pmli) [inline src/pymor/reductors/aaa.py:36]:
```suggestion
        Sampling values are represented as a list of |NumPy arrays| such that `sampling_values[i]` corresponds
```

**Review by pmli — COMMENTED** [review]
It would also be good to make all demo scripts executable (`chmod +x ...`).

**Pull request merged**